### DISCUSSION TITLE
To control led's of a usb webcam

### DISCUSSION THREAD

**Discussion by gauravjagtap123** [pyusb/pyusb#461] [Q&A] [answered]
Hello everyone,

I am using a usb micrscope and i wanted to control the LED's of the micrscope (it's a usb webcam).
I am uisng wireshark to trace the packets sent to the device but i was unable to locate which command/hex value is being sent to the LED's in order to control it.

It will be really helpful if someone tell me how to find the the exact value that is being sent to the leds of the camera in wireshark in order to control it, it will be really helpful.

Thanks in advance

**Comment by jonasmalacofilho** [comment, marked as answer]
First, we're talking about lighting LEDs, right?  (Not merely status/activity LEDs).

I don't know how these microscopes generally work, but I would imagine it's some sort of custom protocol for the LEDs. This means you really need to look into the captured traffic until something clicks, as they could implement this feature in an infinite number of ways. In other words, there's no way we, without having experience with this specific device, can just guess "how to find the exact value".  In fact, while it could be a simple as that, it could also be much much more complicated...

What I can suggest is:

- since the device is mainly a USB webcam, there will be a lot of traffic related to that... try to filter it out;
- in particular, it's somewhat likely that this device exposes more than one USB interface (in which case the LEDs will likely be controlled by a different interface than the one used for the camera);
- alternatively, the device may simply use custom control transfers to control the LEDs, so also look into those in detail;
- and remember that you're likely (even if not surely) looking for very _rare_ packets, that only go out when you change the LED settings in the first party software.

---

P.S. Please don't open issues for questions, especially when those are not specific to PyUSB.  Instead, please use the discussions tab (where I've moved this issue to), or the mailing list.

**Reply by gauravjagtap123** [reply]
Thank you for your feedback.
